### PR TITLE
fix(ui): Fix `<DropdownMenu>` Actor warnings on "Date Selector"

### PR DESCRIPTION
**Pull request by billyvg**
This fixes the `<DropdownMenu>` Actor warnings that happen when you click on the Global Selection Header -> Date selector. This is because we are not forwarding refs in `<HeaderItem>`.

Fixes JAVASCRIPT-21D9

### CODE REVIEW

File: src/sentry/static/sentry/app/components/organizations/headerItem.tsx [hunk 1]
```diff
@@ -20,7 +20,10 @@ type Props = {
   locked: boolean;
   loading: boolean;
   onClear: () => void;
-} & React.HTMLProps<HTMLDivElement>;
+  children: React.ReactNode;
+  className: string;
+  forwardRef?: React.Ref<HTMLDivElement>;
+} & Pick<React.HTMLProps<HTMLDivElement>, 'onClick'>;
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/organizations/headerItem.tsx]: I would typically just write this as `onclick: React.HTMLProps<HTMLDivElement>['onClick']` in the props list.

Personal preference maybe tho.

**Review by evanpurkhiser — APPROVED** [review]

File: src/sentry/static/sentry/app/components/organizations/headerItem.tsx [hunk 2]
```diff
@@ -54,9 +58,10 @@ class HeaderItem extends React.Component<Props> {
       locked,
       lockedMessage,
       settingsLink,
-      onClear, // eslint-disable-line no-unused-vars
+      onClick,
       loading,
-      ...props
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/organizations/headerItem.tsx]: Any concern dropping spread props?

Comment (billyvg) [reply src/sentry/static/sentry/app/components/organizations/headerItem.tsx]: Looks like I missed `data-test-id`, I'll see if I can make it work but ts was blowing up with the spread.

File: src/sentry/static/sentry/app/components/organizations/headerItem.tsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/organizations/headerItem.tsx]: Yeah I think it's just a pref -- with `Pick<>` we could add more props easily.

File: src/sentry/static/sentry/app/components/organizations/headerItem.tsx (hunk 2 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/organizations/headerItem.tsx]: K reverted and changed `HTMLProps` to `HTMLAttributes`

**Pull request merged**